src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1952,3 +1952,23 @@ void TorrentHandle::prioritizeFiles(const QVector<int> &priorities)
 
     updateStatus();
 }
+
+QVector<qreal> BitTorrent::TorrentHandle::filesAvailability() const
+{
+    QVector<int> piecesAvailability = pieceAvailability();
+    const auto filesCount = this->filesCount();
+    if (piecesAvailability.empty()) { // libtorrent returns empty array for seeding only torrents
+        return QVector<qreal>(filesCount, -1.);
+    }
+    QVector<qreal> res;
+    TorrentInfo info = this->info();
+    for (int file = 0; file < filesCount; ++file) {
+        TorrentInfo::PieceRange filePieces = info.filePieces(file);
+        int availablePieces = 0;
+        for (int piece = filePieces.first(); piece <= filePieces.last(); ++piece) {
```

Comment:
I would prefer to leave brackets here in place: with long lines they increase readability, IMO.